mood-analysis.rb

```diff
@@ -19,6 +19,52 @@ def analyze_mood(words)
   return ":-|"
 end
 
+def strip_punctuation(words)
+  words.downcase.gsub(/[^0-9^a-z ]/, '')
+end
+
+def happy_days(day)
+  count = 0
+  day_record = 0
+  num_days_before_happy_three = 0
+  text.each do |aline|
+    count += 1
+    line = strip_punctuation(aline)
+    face = analyze_mood(line)
+    (day_record += 1) if face==":-)"
+    if day_record == 3 && num_days_before_happy_three == 0
+      num_days_before_happy_three = count
+    end
+  end
+  if num_days_before_happy_three == 0
+    puts "WHY ARE YOU SO SAD"
+  else
+    puts "It takes #{num_days_before_happy_three} entries for 3 happy days to occur"
+  end
+end
+
+def overall_mood
+  all = {
+    positive: 0,
+    negative: 0,
+    meh: 0
+  }
+  text.each do |aline|
+    line = strip_punctuation(aline)
+    face = analyze_mood(line)
+    if face == ":-)"
+      all[:positive] +=1
+    if face == ":-("
+      all[:negative] +=1
+    else
+      all[:meh] +=1
+    end
+    end
+  largest = all.max_by{|key, value| value}
+  puts "#{largest.keys} is most common mood"
+end
```

Comment:
This looks really good! And yes, picking the hash was great... Only thing is why rename to positive, negative, and meh, when we could have just used the emoticons themselves. That would have allowed for easy expansion if more feelings got added.

i.e.
```
  emoticons = Hash.new(0)
  entries.each do |entry|
    emoticon = analyze_mood(entry)
    emoticons[emoticon] += 1
  end
```

And then you could have done your same max_by to find the largest.